jenkins/bootstrap_test.py

```diff
@@ -1416,10 +1426,16 @@ def CheckBootstrapYaml(self, path, check, suffix='suffix', use_json=False):
             self.assertTrue(job_name)
             self.assertEquals(job_name, real_job.get('job-name'))
 
+    def GetRealBootstrapJob(self, job):
+        key = os.path.splitext(job.strip())[0][3:]
```

Comment:
Consider refactoring this to check and see whether key is in self.realjobs, returning the cached version if it is, otherwise loading the yaml. Then update CheckBootstrapYaml() to call this.